modules/cloudsql-instance/variables.tf

```diff
@@ -203,8 +203,12 @@ variable "tier" {
 }
 
 variable "users" {
-  description = "Map of users to create in the primary instance (and replicated to other replicas) in the format USER=>PASSWORD. For MySQL, anything afterr the first `@` (if persent) will be used as the user's host. Set PASSWORD to null if you want to get an autogenerated password."
-  type        = map(string)
-  default     = null
+  description = "list of users to create in the primary instance (and replicated to other replicas). For MySQL, anything afterr the first `@` (if persent) will be used as the user's host. Set PASSWORD to null if you want to get an autogenerated password. The user types available are: 'BUILT_IN', 'CLOUD_IAM_USER' or 'CLOUD_IAM_SERVICE_ACCOUNT'"
+  type = list(object({
+    name     = string
+    password = optional(string)
+    type     = string
+  }))
+  default = null
```

Comment:
I kind of like having the username as keys. What do you think of the following type
```hcl
  type = map(object({
    password = optional(string)
    type     = optional(string)
  }))
```

Also, type should be optional and with a validation.

Reply:
ok, I change it asap